### PR TITLE
feat(addon-mobile): more customization for sidebar

### LINKED ISSUE #553: 🚀 - Customizable Sidebar
# 🚀 Feature request

It would be nice if we could customize sidebar drawer like Angular Material sidenav:

https://material.angular.io/components/sidenav/overview#changing-the-sidenavs-behavior

Untitled

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #553

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/186685804-a3b0a681-4497-4200-899c-2d4427654500.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/demo/src/modules/directives/sidebar/examples/2/index.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    placeholder="Choose mode"
+                />
+                <tui-data-list-wrapper
+                    *tuiDataList
+                    [items]="modeState"
+                ></tui-data-list-wrapper>
+            </tui-select>
+
+            <tui-select [formControl]="backdrop">
+                Has backdrop
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/sidebar/examples/2/index.html:23]: nitpicking:

I think selector with true/false values should be replaced with [Toggle](https://taiga-ui.dev/components/toggle) / [Checkbox](https://taiga-ui.dev/components/checkbox).

File: projects/demo/src/modules/directives/sidebar/examples/2/index.less [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.container {
+    width: 43.75rem;
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/sidebar/examples/2/index.less:2]: Please add mobile support
Screenshot 2022-08-26 at 11 26 49

File: projects/demo/src/modules/directives/sidebar/sidebar.template.html [hunk 3]
```diff
@@ -18,6 +18,15 @@
         >
             <tui-sidebar-example-1></tui-sidebar-example-1>
         </tui-doc-example>
+
+        <tui-doc-example
+            id="basic"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/sidebar/sidebar.template.html:23]: It duplicates the first one

File: projects/demo/src/modules/directives/sidebar/examples/2/sidebar/my-sidebar.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    mode: 'over' | 'side' | 'push';
+    hasBackdrop: boolean;
+}
+
+@Component({
+    selector: `my-sidebar`,
+    templateUrl: `./my-sidebar.component.html`,
+    styleUrls: [`./my-sidebar.component.less`],
+})
+export class MySidebarComponent implements TuiSidebar {
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/sidebar/examples/2/sidebar/my-sidebar.component.ts:15]:
```ts
import {changeDetection} from '@demo/emulate/change-detection';
import {encapsulation} from '@demo/emulate/encapsulation';
```

**Comment by Blockost** [comment]
@splincode Any reason why this PR has been closed ? I'm interested in having more ways to customize the sidebar 🤞

**Pull request closed without merging**